### PR TITLE
New raster upload cmd

### PR DESCRIPTION
**Pull request by pratikyadav**
Testing alternative approach to https://github.com/mapbox/tilesets-cli/pull/183 

This PR adds a new command `tilesets upload-raster-source` to upload raster file to MTS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +119,21 @@ tilesets upload-source <username> <source_id> file-1.geojson file-4.geojson
 # Reading from a directory will not distinguish between GeoJSON files and non GeoJSON files. All source files will be run through our validator unless you pass the `--no-validation` flag.
 tilesets upload-source <username> <source_id> ./path/to/multiple/files/
 ```
+### upload-raster-source
+
+```shell
+tilesets upload-source-source <username> <source_id> <file>
+```
+
+Uploads Raster files to a source for tiling. Accepts GeoTIFF, NetCDF and GRIB formats right now. Can be used to add data to a source or to replace all of the data in a source with the `--replace` flag.
```
Comment (greenlaw) [inline README.md:128]: I don't think there's a way to append raster data, so it seems like `--replace` is unneeded (raster source will always replace previous one)

File: mapbox_tilesets/scripts/cli.py [hunk 2]
```diff
@@ -629,6 +630,109 @@ def callback(m):
...
+    tilesets upload-source <username> <source_id> <path/to/source/data>
+    """
+    return _upload_raster_source(
+        ctx, username, id, inputs, quiet, replace, token, indent
+    )
+
+
+def _upload_raster_source(
+    ctx, username, id, inputs, quiet, replace, token=None, indent=None
+):
```
Comment (greenlaw) [inline mapbox_tilesets/scripts/cli.py:661]: In order to support multiple `inputs`, I think you would also need to support multiple `id` values (i.e. can't have more than one raster per source id, right?).  If that's the case, may want to only support one file upload at a time for raster.

Comment (greenlaw) [reply mapbox_tilesets/scripts/cli.py:661]: Nevermind!

File: README.md [hunk 3]
```diff
@@ -118,6 +119,21 @@ tilesets upload-source <username> <source_id> file-1.geojson file-4.geojson
 # Reading from a directory will not distinguish between GeoJSON files and non GeoJSON files. All source files will be run through our validator unless you pass the `--no-validation` flag.
 tilesets upload-source <username> <source_id> ./path/to/multiple/files/
 ```
+### upload-raster-source
+
+```shell
+tilesets upload-source-source <username> <source_id> <file>
```
[the remark this answers is not in the document]
Comment (pratikyadav) [reply README.md]: 👍 edited.

**Review by greenlaw — APPROVED** [review]
Looks good 👍

**Review by greenlaw — APPROVED** [review]

**Review by jqtrde — APPROVED** [review]
Looks great, thanks @pratikyadav @greenlaw

**Pull request merged**